app/Http/Controllers/AmaController.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\DB;
+
+class AmaController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index(Request $request)
+    {
+       
+   
+         $amas = DB::table('amas')->select('title', 'person', 'tags', 'created_at')->paginate(20);
+
+        
+        echo $request->date_from;
+        echo $request->date_to;
+        echo $request->tags;
+        
+            if ($request->has('date_from')) {
+
+
+               $amas = DB::table('amas')->select('title', 'person', 'tags', 'created_at')->whereBetween('created_at', [$request->date_from, $request->date_to])->paginate(20);
```

Comment:
This way you have a separate query for every filter, which means you will never be able to have the items filtered by both date and tags.

1. Since `$amas` is a created object already, use it to add all filters below. Remove the paginate from line 19 and then in every if, add the condition like so: `$amas->whereBetween` instead of replacing the whole query.
2. You're missing a check for `date_to`. It also needs to be included. Right now, if date_to is empty, the `whereBetween` won't work. Add another if for this and check it like this:

```
if($request->date_from) {
     $amas->where('created_at', '>=', $request->date_from)
}
if($request->date_to) {
     $amas->where('created_at', '<=', $request->date_to)
}
```